### DISCUSSION TITLE
RetroNET NABU Internet Adapter (show & tell)

### DISCUSSION THREAD

**Discussion by DJSures** [tui-cs/Terminal.Gui#2351] [Show and tell]
I wanted to share this app I built using Terminal.Gui for the NABU RetroNET project (https://nabu.ca). Quick info about the NABU: it was a Canadian computer from 1980-1984 that was entirely networked over coax. The vision was to have a computer that connected people and communities - even to pay bills and order flowers. The television cable companies did not agree and never upgraded their infrastructure for bi-directional communication, leaving the computer only to receive data. The "hack" was to broadcast ALL available software on a carousel wheel. This meant all software would be broadcasted and repeated - the NABU computer's network adapter would pick up the desired program and load it. I have reverse-engineered the protocol, obtained the original software, and created a program that simulates the network broadcast server. I named it the NABU Internet Adapter. For true cross-platform single-build one-codebase, I choose Terminal.GUI, and it's going great. There are only 1800 nabu's available on earth, so that's the usage cap for the software, but it still has a good user base.

The software allows users to select a source for the data. I've also imagined what the NABU development would look like to alter the protocol for bi-directional communication since it never got done. I've added a remote filesystem, TCP stream, chat, remote resources, and a bunch of other stuff to extend the existing protocol. I even created a custom NABU BIOS with a remote filesystem called Cloud CP/M, so the computer doesn't require local storage - it's all done on the Terminal.GUI program.

Here's a screenshot
![image](https://user-images.githubusercontent.com/49293683/219811751-d60d96e9-ccce-4218-bb0f-9d1df80954d1.png)

**Comment by tznind** [comment]
Oh man this is so awesome!!!

Thanks so much for sharing this.  I was able to get the Internet Adapter running and the MAME emulator running under WINE.  And got rick rolled by the DOOM channel haha.

Very awesome stuff! Looks like a ton of work went into this! I'm looking forward to playing more with this.

![NABUWorking](https://user-images.githubusercontent.com/31306100/219818031-4fb928db-0d29-4fee-8202-845ce6c20f32.jpg)

**Reply by DJSures** [reply]
HAHAHA, That's so great! I appropriately named that rickroll! People on YouTube kept asking when it would run doom after I got bad apple going. Well - there you go!

**Reply by tig** [reply]
This. Is. Awe. Inspiring. 

Thank you for sharing!

**Reply by DJSures** [reply]
thank you so much - that is really kind of you to say!

**Comment by tig** [comment]
I'd like to interview you on your Terminal.gui experience. We're embarking on v2 and i want to make sure we're considering the right improvements. You game?

**Comment by DJSures** [comment]
I want to help however I can - it's a great tool. I'm unsure how useful I'll be, but I'm game.
